lib/statsd.js

```diff
@@ -583,4 +587,5 @@ function trySetNewSocket(client) {
     protocol: client.protocol,
     stream: client.stream,
   });
+  client.lastSocketCreateTime = Date.now();
```

Comment:
Should I move this into `createTransport`?

Reply:
IMO it's critical (pending discussion of whether it makes sense to bloat the interface of `client`)

Reply:
I actually meant moving the assignment of `Date.now()`, but maybe we can move the field to `client.socket` and set it in `createTransport` like we do the `type` field.
https://github.com/brightcove/hot-shots/blob/a64057a6c4f02b5ed38ddd4249854571f0412864/lib/transport.js#L160